File: csharp/qfroms/Queue.cs
using System;
using System.Collections;

namespace Qfroms
{
    /// <summary>
    /// --- Directions
    /// Implement a Queue datastructure using two stacks.
    /// *Do not* create an array inside of the 'Queue' class.
    /// Queue should implement the methods 'add', 'remove', and 'peek'.
    /// For a reminder on what each method does, look back
    /// at the Queue exercise.
    /// </summary>
    /// <example>
    /// var q = new Queue();
    /// q.Add(1);
    /// q.Add(2);
    /// q.Peek();   // returns 1
    /// q.Remove(); // returns 1
    /// q.Remove(); // returns 2
    /// <example>
    public class Queue
    {
        private Stack left;
        private Stack right;

        public Queue()
        {
            this.left = new Stack();
            this.right = new Stack();
        }

        public void Add(object element)
        {
            while (this.right.Peek() != null)
            {
                this.left.Push(this.right.Pop());
            }

            this.left.Push(element);
        }

        public object Remove()
        {
            while (this.left.Peek() != null)
            {
                this.right.Push(this.left.Pop());
            }

            return this.right.Pop();
        }

        public object Peek()
        {
            while (this.left.Peek() != null)
            {
                this.right.Push(this.left.Pop());
            }

            return this.right.Peek();
        }
    }
}

File: php/qfroms/Queue.php
<?php declare(strict_types=1);

/**
 * --- Directions
 * Implement a Queue datastructure using two stacks.
 * *Do not* create an array inside of the 'Queue' class.
 * Queue should implement the methods 'add', 'remove', and 'peek'.
 * For a reminder on what each method does, look back
 * at the Queue exercise.
 *
 * @example
 *     $q = new Queue();
 *     $q.add(1);
 *     $q.add(2);
 *     $q.peek();   // returns 1
 *     $q.remove(); // returns 1
 *     $q.remove(); // returns 2
 */
class Queue
{
    private $left;
    private $right;

    public function __construct()
    {
        $this->left = new Stack();
        $this->right = new Stack();
    }

    public function add(int $element): void
    {
        while ($this->right->peek() !== null) {
            $this->left->push($this->right->pop());
        }

        $this->left->push($element);
    }

    public function remove(): ?int
    {
        while ($this->left->peek() !== null) {
            $this->right->push($this->left->pop());
        }

        return $this->right->pop();
    }

    public function peek(): ?int
    {
        while ($this->left->peek() !== null) {
            $this->right->push($this->left->pop());
        }

        return $this->right->peek();
    }
}